### PR TITLE
refactor(connector): [Forte] Remove Default Case Handling

### PR DESCRIPTION
**Pull request by SagarDevAchar**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Addresses Issue #2270 
- Modified `crates/router/src/connector/forte/transformers.rs`
  - Convert `NotImplemented` to `NotSupported` in default case

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/forte/transformers.rs [hunk 1]
```diff
@@ -101,9 +101,10 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for FortePaymentsRequest {
                     card,
                 })
             }
-            _ => Err(errors::ConnectorError::NotImplemented(
-                "Payment Method".to_string(),
-            ))?,
+            _ => Err(errors::ConnectorError::NotSupported {
```
Comment (Sangamesh26) [inline crates/router/src/connector/forte/transformers.rs:104]: Instead of default handling('_'), can you expand all the cases and assign it to Error

**Review by Sangamesh26 — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/forte/transformers.rs (hunk 1 above)
Comment (SagarDevAchar) [reply crates/router/src/connector/forte/transformers.rs]: Updated!

**Review by Sangamesh26 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @SagarDevAchar!

**Comment by swangi-kumari** [comment]
Hello @SagarDevAchar ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**